docs/templates.md

```diff
@@ -0,0 +1,130 @@
+# Armada Spark Job Templates
+
+## Introduction
+
+Job templates were introduced to provide a reusable way to configure Spark jobs on Armada with common settings, while still allowing runtime customization.
+This feature addresses the need for standardizing Armada job configurations across teams and environments.
+
+## How It Works
+
+The job template system uses a precedence hierarchy where CLI configuration flags override template values.
+Templates define base configurations for driver and executor pods in YAML format, which are then merged with runtime settings.
+
+Templates can be loaded from:
+- Local files (absolute or relative paths, with or without `file://` prefix)
+- HTTP/HTTPS URLs
+
+## Configuration Options
+
+### Spark Configuration Properties
+
+```properties
+# Job-level template (applies to both driver and executors)
+spark.armada.jobTemplate=/path/to/job-template.yaml
+
+# Driver-specific template
+spark.armada.driver.jobItemTemplate=/path/to/driver-template.yaml
+
+# Executor-specific template
+spark.armada.executor.jobItemTemplate=/path/to/executor-template.yaml
+```
+
+### Using Templates
+
+Templates are specified via `--conf` flags when submitting Spark jobs:
+
+```bash
+spark-submit \
+  --master armada://armada-server:50051 \
+  --conf spark.armada.jobTemplate=https://config-server/templates/base-job.yaml \
+  --conf spark.armada.driver.jobItemTemplate=/templates/driver.yaml \
+  --conf spark.armada.executor.jobItemTemplate=/templates/executor.yaml \
+  --conf spark.armada.scheduling.queue=gpu-queue \
+  --conf spark.armada.scheduling.priority=100 \
+  --class com.example.SparkApp \
+  app.jar
+```
+
+## Template Examples
+
+### Job Template (JobSubmitRequest)
```

Comment:
Someone reading this doc may not know what "JobSubmitRequest" means. How about "Armada JobSubmitRequest Api"?

Reply:
done

Reply:
can you push the changes so I can see them?